flytekit/core/map_task.py

```diff
@@ -93,9 +97,19 @@ def get_container(self, settings: SerializationSettings) -> Container:
             environment=env,
         )
 
+    def get_k8s_pod(self, settings: SerializationSettings) -> K8sPod:
+        if self._run_task.task_type not in _K8S_POD_TARGET_TASK_TYPES:
+            return None
+
+        self._run_task.set_command_fn(self.get_command)
```

Comment:
Struggled with the right way to do this. An alternative was serializing the run_task's k8s pod as is and updating the command with `pyflyte-map-execute` after the fact but then that requires the map task to be familiar with the serialized structure of each non-container target type task, which didn't feel right.

Another option to avoid the _K8S_POD_TARGET_TASK_TYPES set is to call the corresponding method, (e.g. `_run_task.get_k8s_pod(settings)` and then see if it's empty or not before returning in this method - but this seems (marginally) inefficient.

If you can think of a better way to do this, let me know

Reply:
I think it is fair to call https://github.com/flyteorg/flytekit/blob/master/plugins/pod/flytekitplugins/pod/task.py#L103. This is because we now have K8sPodSpec as a target.
So ideally we should have one method per target. Or we could simply call it getTarget

Reply:
the only thing I would say is it should be at the base level. the base should have getContainer, getK8sPodSpec, getSQL, getHTTPAPI etc
and the semantics should be crisp and state that we will only invoke one of them.

Now the question is which one to invoke?

Reply:
I think for python proto serialization we can't just produce one getTarget method for the oneof field, unfortunately. So each task needs to implement each get_foo target option

re: invoking each core target and having each return None except for the applicable target - this sounds good. Refactored, PTAL @kumare3